src/commands/cereal/hackernews.ts

```diff
@@ -0,0 +1,135 @@
+import { Message, MessageEmbed } from "discord.js";
+import _ from "lodash";
+import fetch from "node-fetch";
+import { HACKERNEWS as H, DISCORD } from "../../../config.json";
+import logger from "../../logging";
+import { HackerNews } from "../../models/hackerNews";
+import { getCerealColor, trim } from "../../utils";
+
+const fetchHackerNews = async (url: string): Promise<Object> => {
+  return await fetch(url, {}).then(response => response.json());
+};
+
+const checkValidResponseOfTopHackerNews = (res: Object): number[] => {
+  if (_.isArrayLike(res)) {
+    if (_.isInteger(res[0])) {
+      return res as number[];
+    }
+  }
+  throw new Error(`HackerNews response is not parsable:\n${res}`);
+};
+
+const takeTopPosts = (res: number[], numberOfPosts: number): number[] => {
+  return res.slice(0, numberOfPosts);
+};
+
+/**
+ * Get news stories based on a list of story ids
+ * @param stories Array of ids of stories
+ * @returns Array of HackerNews
+ */
+const getNewsStories = async (stories: number[]): Promise<HackerNews[]> => {
+  const baseURL = H.BASE_URL;
+  let listOfNews = [];
+  for (const i of stories) {
+    const s = await fetchHackerNews(baseURL + H.STORY + i + H.URL_SUFFIX);
+    const newsStory = new HackerNews(
+      s["by"],
+      s["descendants"],
+      s["id"],
+      s["kids"],
+      s["score"],
+      s["time"],
+      s["title"],
+      s["type"],
+      s["url"]
+    );
+    listOfNews.push(newsStory);
+  }
+  return listOfNews;
+};
+
+/**
+ * Parse args. The args can only be integers and 0 < x < 11
+ * @param args arguments of the command
+ * @returns Parsed args to get the number of posts to create
+ */
+const parseArgs = (args: string[]): number => {
+  let numberOfPosts = 5;
+
+  if (args) {
+    if (args.length === 1) {
+      const parsedArg = parseInt(args[0]);
+      if (_.isInteger(parsedArg)) {
+        numberOfPosts = Math.abs(parsedArg);
+        if (numberOfPosts >= 10) {
+          numberOfPosts = 10;
+        }
```

Comment:
Could be rewritten:
numberOfPosts = Math.max(parsedArg, 0);
numberOfPosts = Math.min(parsedArg, 11);